src/api/group-role.ts

```diff
@@ -0,0 +1,60 @@
+import { PulpAPI } from './pulp';
+import { RoleAPI } from './role';
+
+class API extends PulpAPI {
+  apiPath = 'groups/';
+
+  constructor() {
+    super();
+  }
+
+  async getRolesWithPermissions(id, params?) {
+    const assignedRoles = await this.list(
+      params,
+      this.apiPath + `${id}/roles/`,
+    );
+
+    const allRoles = await RoleAPI.list();
+
+    return new Promise((resolve, reject) => {
+      Promise.all([assignedRoles, allRoles])
+        .then(([assigned, all]) => {
+          // match roles with assigned roles
+          const data = assigned.data.results
+            .map(({ role, pulp_href }) => {
+              const data = all['data'].results.find(
+                ({ name }) => name === role,
+              );
+              if (data) {
+                return {
+                  ...data,
+                  // swap pulp_href role with assigned pulp_href role
+                  // to delete the assigned role
+                  pulp_href,
+                };
+              }
+            })
+            .filter(Boolean);
+
+          resolve({
+            data,
+            count: data.length,
+          });
+        })
+        .catch((e) => reject(e));
+    });
+  }
```

Comment:
this is a surprising combination of async and promise, and is waiting in places it probably shouldn't

```suggestion
  getRolesWithPermissions(id, params?) {
    const assignedRoles = this.list(
      params,
      this.apiPath + `${id}/roles/`,
    );

    const allRoles = RoleAPI.list();

    return Promise.all([assignedRoles, allRoles])
      .then(([assigned, all]) => {
        // match roles with assigned roles
        const data = assigned.data.results
          .map(({ role, pulp_href }) => {
            const data = all['data'].results.find(
              ({ name }) => name === role,
            );
            if (data) {
              return {
                ...data,
                // swap pulp_href role with assigned pulp_href role
                // to delete the assigned role
                pulp_href,
              };
            }
          })
          .filter(Boolean);

        return {
          data,
          count: data.length,
        };
      })
  }
```